ddl/rollingback.go

```diff
@@ -175,11 +199,35 @@ func rollingbackDropColumn(t *meta.Meta, job *model.Job) (ver int64, err error)
 }
 
 func rollingbackDropColumns(t *meta.Meta, job *model.Job) (ver int64, err error) {
-	tblInfo, colInfos, _, err := checkDropColumns(t, job)
+	tblInfo, colInfos, _, idxInfos, err := checkDropColumns(t, job)
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
 
+	if len(idxInfos) > 0 {
+		for _, indexInfo := range idxInfos {
+			originalState := indexInfo.State
+			switch indexInfo.State {
+			case model.StateWriteOnly, model.StateDeleteOnly, model.StateDeleteReorganization, model.StateNone:
+				// We can not rollback now, so just continue to drop index.
+				// In function isJobRollbackable will let job rollback when state is StateNone.
+				// When there is no index related to the drop columns job it is OK, but when there has indices, we should
+				// make sure the job is not rollback.
+				job.State = model.JobStateRunning
+				return ver, nil
+			case model.StatePublic:
+				indexInfo.State = model.StatePublic
+			default:
+				return ver, ErrInvalidDDLState.GenWithStackByArgs("index", indexInfo.State)
+			}
```

Comment:
L223~227 can be eliminated since tableInfo hasn't been changed in the index state (public to public), we don't have to set it again. 

the logic here is: check the index is public, then the column is public, both public means the drop job hasn't run even for the once(tableInfo hasn't been modified even for the once), so we can change the job to  JobStateCancelled without doing anything with the tableinfo.

Reply:
@AilinKid do you mean at L232 we should check colInfos[0].State and all IndexInfo.State is StatePublic then we can rollback it?

Reply:
Removing the L223~227 is adequate